scene/gui/tab_container.cpp

```diff
@@ -417,7 +419,13 @@ void TabContainer::set_current_tab(int p_current) {
 	}
 
 	_change_notify("current_tab");
-	emit_signal("tab_changed", current);
+	if (pending_previous == current)
+		emit_signal("tab_selected", current);
```

Comment:
It does change,  I tested exactly what you suggested, on all subsequent calls before the emission occurs, `pending_previous` changes to `current` at the start of the function. Which ruins the comparison later on. Note that I don't compare previous with current; but pending previous with current.

I find the current code comprehendible and neat.